### PR TITLE
Create configuration setup

### PR DESCRIPTION
**Pull request by SaranjeetKaur**
Refactor data-process.R using the config file

### CODE REVIEW

**Comment by SaranjeetKaur** [comment]
@jzhu20 - please let me know what you think about the refactored gen_data_weighted function. I have used the config file to rewrite this function.

File: inst/config/config.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+default:
+  grouping_vars:  # Variables that are used to group the data
+    - simID
+    - source
+    - time
+  weight:
```
Comment (jzhu20) [inline inst/config/config.yml:6]:
```suggestion
  weight:  # Population size of each group is used as weight
```

File: inst/config/config.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    - count
+  weighted_vars:
```
Comment (jzhu20) [inline inst/config/config.yml:8]:
```suggestion
  weighted_vars:  # To calculate population weighted mean
```

File: inst/config/config.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - fat
+    - migrations
+    - obesity
+    - overweight
+    - physicalactivity
+    - protein
+    - sector
+    - sodium
+    - weight
+  prevalence:
```
Comment (jzhu20) [inline inst/config/config.yml:25]:
```suggestion
  prevalence:  # List of disease to calculate the total prevalence number in the population
```

Comment (jzhu20) [reply inst/config/config.yml:25]: Name this list as disease might be more intuitive.

File: inst/config/config.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    - asthma
+    - ckd
+    - diabetes
+    - ihd
+    - stroke
+  stroke:
+    - intracerebralhemorrhage
+    - ischemicstroke
+    - subarachnoidhemorrhage
+  total:
```
Comment (jzhu20) [inline inst/config/config.yml:35]:
```suggestion
  burden:  # List of disease burden measures 
```

Comment (jzhu20) [reply inst/config/config.yml:35]: burden might be a more suitable name of this list.

**Pull request merged**